karma_chameleon.py

```diff
@@ -0,0 +1,140 @@
+# by all sub-loggers.
+logger = logging.getLogger("karma_chameleon")
+logger.setLevel(logging.DEBUG)
+file_handler = RotatingFileHandler(
+    "logs/karma_chameleon.log", maxBytes=10_000_000, backupCount=3
+)
+file_handler.setLevel(logging.DEBUG)
+formatter = logging.Formatter(
+    "%(asctime)s %(name)s %(levelname)s[%(funcName)s]: %(message)s"
+)
+file_handler.setFormatter(formatter)
+logger.addHandler(file_handler)
+
+app = KarmaBot(
+    token=os.environ.get("SLACK_BOT_TOKEN"),
+    signing_secret=os.environ.get("SLACK_SIGNING_SECRET"),
+)
+
+
+@app.middleware
+def handle_no_karma_op(
+    body: dict, next: Callable  # pylint: disable=redefined-builtin
+) -> Any:
+    """Middleware which enables KarmaChameleon to immediately and gracefully handle events which do
+    not contain any karma operations or slash-commands.
+
+    Unfortunately the Slack Bolt API requires that the next middleware method be referred to as
+    "next", which is a built-in.  You can't win them all I suppose...
+
+    Arguments:
+    body -- a dict containing the entire Slack Bolt API event
+    next_middleware -- callable reference to the next middleware listener.
+
+    Returns:
+    If the incoming event contains a karma operation, then the next middleware listener is returned
+    and invoked by the caller of this middleware.  If there is no karma operation contained in the
+    event, and the event is not a command, then a BoltResponse(200) is returned as we have nothing
+    further to do.
+    """
+    if body.get("command"):
+        return next()
+
+    if body["event"]["type"] == "message":
+        msg = body["event"]["text"]
+        if app.inc_regex.match(msg) or app.dec_regex.match(msg):
+            return next()
+    # This is too chatty to be left enabled, but it may be useful for debug in the future.
+    # logger.debug("Ignoring event with no karma operation")
+    return BoltResponse(status=200, body="Ignoring event with no karma operation")
+
+
+@app.message(app.inc_regex)
+def increment(message: dict, say: Say) -> None:
+    """Passes the message along for to the KarmaChameleon bot, then posts a response based on the
+    return value of the bot's increment method.
+
+    Arguments:
+    message -- dictionary representation of the message which contains a karma operation
+    say -- method for printing back to the same channel from which the command was run
+    """
+    rsp = app.increment_karma(message)
+    say(rsp)
+
+
+@app.message(app.dec_regex)
+def decrement(message: dict, say: Say) -> None:
+    """Passes the message along for to the KarmaChameleon bot, then posts a response based on the
+    return value of the bot's decrement method.
+
+    Arguments:
+    message -- dictionary representation of the message which contains a karma operation
+    say -- method for printing back to the same channel from which the command was run
+    """
+    rsp = app.decrement_karma(message)
+    say(rsp)
+
+
+@app.command("/ec_leaderboard")
```

Comment:
This needs to be put back... The command present now is debug I forgot to remove.